charts/partials/templates/_proxy.tpl

```diff
@@ -95,7 +95,7 @@ env:
   value: "true"
 {{ else if not .Values.global.proxy.disableIdentity -}}
 - name: LINKERD2_PROXY_TAP_SVC_NAME
-  value: linkerd-tap.$(_l5d_ns).serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)
+  value: linkerd-tap.{{.Values.linkerdVizNamespace}}.serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)
```

Comment:
This logic will eventually move to the tap-injector in the linkerd-viz extension.  For now I'd say just hardcode the namespace.